src/api/app/models/workflow/step/configure_repositories.rb

```diff
@@ -15,8 +17,10 @@ def call(_options = {})
     step_instructions[:repositories].each do |repository_instructions|
       repository = Repository.includes(:architectures).find_or_create_by(name: repository_instructions[:name], project: target_project)
 
-      target_repository = Repository.find_by_project_and_name(repository_instructions[:target_project], repository_instructions[:target_repository])
-      repository.path_elements.find_or_create_by(link: target_repository)
+      repository_instructions[:paths].each do |repository_path|
+        target_repository = Repository.find_by_project_and_name(repository_path[:target_project], repository_path[:target_repository])
```

Comment:
Any possibility to this call return `nil`?

Reply:
Yes, this implementation is somewhat naive, hoping that users provide an existing repository. If `target_project` and `target_repository` do not refer to an existing repository, users won't be informed about this.

I'll see what I can do to improve this.

Reply:
Are you going to do this in this PR?

Reply:
I've created a [card](https://trello.com/c/MMyx4rDc/2489-validate-existence-of-repositories-in-configurerepositories) for our backlog, so I won't tackle this in this PR.